src/Sylius/Bundle/CoreBundle/DependencyInjection/Compiler/CircularDependencyBreakingErrorListenerPass.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\CoreBundle\DependencyInjection\Compiler;
+
+use Sylius\Bundle\CoreBundle\EventListener\CircularDependencyBreakingErrorListener;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Definition;
+use Symfony\Component\DependencyInjection\Reference;
+
+final class CircularDependencyBreakingErrorListenerPass implements CompilerPassInterface
+{
+    public function process(ContainerBuilder $container)
+    {
+        if ($container->has('exception_listener')) {
+            $definition = new Definition(CircularDependencyBreakingErrorListener::class);
+            $definition->setDecoratedService('exception_listener');
+            $definition->AddArgument(new Reference('exception_listener'));
```

Comment:
```suggestion
            $definition->addArgument(new Reference('exception_listener'));
```

Reply:
```suggestion
            $definition->AddArgument(new Reference(CircularDependencyBreakingErrorListener::class .'.inner'));
```

I'm not sure what that changes, but that's the recommended way to inject decorated services.